src/main/java/com/mineboundteam/minebound/entity/RockProjectile.java

```diff
@@ -0,0 +1,129 @@
+        this.setOwner(shooter);
+
+        this.damage = damage;
+        this.originalX = pX;
+        this.originalY = pY;
+        this.originalZ = pZ;
+        this.maxDist = maxDistance;
+        this.damageRadius = damageRadius;
+        this.setSoundEvent(SoundEvents.STONE_BREAK);
+        //Overwrite damage that would have been applied in AbstractArrow
+        this.setBaseDamage(0.0D);
+    }
+
+    @Override
+    protected void onHitEntity(EntityHitResult pResult) {
+        super.onHitEntity(pResult);
+        if(!this.level.isClientSide){
+            Entity target = pResult.getEntity();
+            Entity player = this.getOwner();
+            target.hurt(DamageSource.thrown(this, player), (float) damage);
+            if (player instanceof LivingEntity p ) {
+                this.doEnchantDamageEffects(p, target);
+            }
+        }
+    }
+
+    @Override
+    protected void onHitBlock(BlockHitResult pResult) {
+        super.onHitBlock(pResult);
+        if (!this.level.isClientSide) {
+            Entity entity = this.getOwner();
+            if (!(entity instanceof Mob) || net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(this.level, this)) {
+                BlockPos blockpos = pResult.getBlockPos().relative(pResult.getDirection());
+                if (this.level.isEmptyBlock(blockpos)) {
+                    List<Entity> hitEntities = this.level.getEntities(this, this.getBoundingBox().inflate(damageRadius));
+                    for (Entity target : hitEntities) {
+                        if (target instanceof LivingEntity) {
+                            target.hurt(DamageSource.thrown(this, entity), (float) damage);
+                            if (entity instanceof LivingEntity p) {
+                                this.doEnchantDamageEffects(p, target);
+                            }
+                        }
+                    }
+                }
+            }
+            this.discard();
+        }
+    }
+
+    @Override
+    protected ItemStack getPickupItem() {
+        return null;
+    }
+
+    @Override
+    public void tick() {
+        super.tick();
+
+        //Add Trail
+        Vec3 vec3 = this.getDeltaMovement();
+        double d5 = vec3.x;
+        double d6 = vec3.y;
+        double d1 = vec3.z;
+        for(int i = 0; i < 4; ++i) {
+            this.level.addParticle(ParticleTypes.ASH, this.getX() + d5 * (double)i / 4.0D, this.getY() + d6 * (double)i / 4.0D, this.getZ() + d1 * (double)i / 4.0D, -d5, -d6 + 0.2D, -d1);
+        }
+
+        if (!this.level.isClientSide()) {
+            double dX = Math.abs(this.getX() - originalX);
+            double dY = Math.abs(this.getY() - originalY);
+            double dZ = Math.abs(this.getZ() - originalZ);
+            if (dX + dY + dZ > maxDist) {
+                this.discard();
+            }
+        }
+    }
+    @Override
+    protected Entity.MovementEmission getMovementEmission() {
+        return MovementEmission.EVENTS;
+    }
```

Comment:
Is this a required override or something you chose to do? If the case is the latter, what is the purpose of it?

Reply:
Abstract defaults to arrow sounds and I wanted to remove said sounds.